contract-examples/contracts/ExampleFeeManager.sol

```diff
@@ -84,3 +104,131 @@ contract ExampleFeeManager is AllowList {
     return feeManager.getFeeConfigLastChangedAt();
   }
 }
+
+contract ExampleFeeManagerTest is DSTest {
+  uint256 testNumber;
+
+  function setUp() public {
+    // noop
+  }
+
+  function test_addContractDeployerAsOwner() public {
+    ExampleFeeManager manager = new ExampleFeeManager();
+    assertEq(address(this), manager.owner());
+  }
+
+  function test_enableWAGMIFeesFailure() public {
+    ExampleFeeManager example = new ExampleFeeManager();
+
+    IFeeManager manager = IFeeManager(FEE_MANAGER_ADDRESS);
+
+    // TODO: make roles const (role: None = 0)
+    assertEq(manager.readAllowList(address(example)), 0);
+
+    try example.enableWAGMIFees() {
+      assertTrue(false, "enableWAGMIFees should fail");
+    } catch {} // TODO should match on an error to make sure that this is failing in the way that's expected
+  }
+
+  function test_addContractToManagerList() public {
+    ExampleFeeManager example = new ExampleFeeManager();
+
+    address exampleAddress = address(example);
+    address thisAddress = address(this);
+
+    IFeeManager manager = IFeeManager(FEE_MANAGER_ADDRESS);
+
+    // TODO: make this a const (role: ADMIN = 2)
+    assertEq(manager.readAllowList(thisAddress), 2);
+    assertEq(manager.readAllowList(exampleAddress), 0);
+
+    manager.setEnabled(exampleAddress);
+
+    // TODO: make this a const (role: ENABLED = 1)
+    assertEq(manager.readAllowList(exampleAddress), 1);
+  }
+
+  function test_changeFees() public {
+    ExampleFeeManager example = new ExampleFeeManager();
+    address exampleAddress = address(example);
+
+    IFeeManager manager = IFeeManager(FEE_MANAGER_ADDRESS);
+
+    manager.setEnabled(exampleAddress);
+
+    FeeConfig memory config = example.getCurrentFeeConfig();
+
+    FeeConfig memory newFeeConfig = FeeConfig({
+      gasLimit: CCHAIN_GAS_LIMIT,
+      targetBlockRate: CCHAIN_TARGET_BLOCK_RATE,
+      minBaseFee: CCHAIN_MIN_BASE_FEE,
+      targetGas: CCHAIN_TARGET_GAS,
+      baseFeeChangeDenominator: CCHAIN_BASE_FEE_CHANGE_DENOMINATOR,
+      minBlockGasCost: CCHAIN_MIN_BLOCK_GAS_COST,
+      maxBlockGasCost: CCHAIN_MAX_BLOCK_GAS_COST,
+      blockGasCostStep: CCHAIN_BLOCK_GAS_COST_STEP
+    });
+
+    assertNotEq(config.gasLimit, newFeeConfig.gasLimit);
+    // target block rate is the same for wagmi and cchain
+    // assertNotEq(config.targetBlockRate, newFeeConfig.targetBlockRate);
+    assertNotEq(config.minBaseFee, newFeeConfig.minBaseFee);
+    assertNotEq(config.targetGas, newFeeConfig.targetGas);
+    assertNotEq(config.baseFeeChangeDenominator, newFeeConfig.baseFeeChangeDenominator);
+    // min block gas cost is the same for wagmi and cchain
+    // assertNotEq(config.minBlockGasCost, newFeeConfig.minBlockGasCost);
+    assertNotEq(config.maxBlockGasCost, newFeeConfig.maxBlockGasCost);
+    assertNotEq(config.blockGasCostStep, newFeeConfig.blockGasCostStep);
```

Comment:
should we instead use the `WAGMI_*`constants as expected values and assertEq instead?
(same below with `CCHAIN_*` constants for the expected values after updating fees?)

Reply:
Ah, the `assertNotEq `assertions here are actually just making sure that _current config_ doesn't have any of the same values as `newFeeConfig`. Changing the actual fees happens below this, then we verify with `assertEq` that all the fees were changed to exactly what was expected.